scripts/bootloader/provision.py

```diff
@@ -63,7 +69,11 @@ def get_hashes(public_key_files):
     hashes = list()
     for fn in public_key_files:
         with open(fn, 'rb') as f:
-            hashes.append(sha256(VerifyingKey.from_pem(f.read()).to_string()).digest()[:16])
+            digest = sha256(VerifyingKey.from_pem(f.read()).to_string()).digest()[:16]
+            if digest_contains_ffff(digest):
+                raise RuntimeError("Hash of key in '%s' contains 0xffff. Please regenerate the key." %
+                                   os.path.abspath(f.name))
```

Comment:
why not do:
```suggestion
            if digest.find(b'\xff\xff\xff\xff') != -1:
                raise RuntimeError("Hash of key in '%s' contains 0xffff. Please regenerate the key." %
                                   os.path.abspath(f.name))
```

Reply:
We're looking for half-word aligned 0xFFFFs

Reply:
right, my suggestion would also fail if the `0xffff` is not half-word aligned.
Question is if that is a real issue. A hash with `0xffff` starting at an odd mem address is just as unlikely as the `0xffff` on a half-word aligned address.

Reply:
I don't see a problem with the current solution, so why willingly introduce false negatives?

Reply:
there is nothing wrong.
Just found it nice if we could safe some lines :wink: 

As alternative, one could do:
```suggestion
            if [n for n in range(8) if digest[2*n:2*n+2] == b'\xff\xff']:
```

which gives you what you want, a check on each half-word aligned 2-bytes.

Reply:
Haha, nice :). maybe like this?

```suggestion
            if any([digest[n:n+2] == b'\xff\xff' for n in range(0, len(digest), 2)]):
                raise RuntimeError("Hash of key in '%s' contains 0xffff. Please regenerate the key." %
                                   os.path.abspath(f.name))
```

Reply:
I also like that one, but a couple of chars longer (and thus violating PEP a bit more than my version :grin: ).
but both are still shorter than the `raise RuntimeError` line.

Happy to approve both versions.